Feature/issue 15 email verification iyte

UlasGokkaya opened the pull request.

UlasGokkaya committed:
feat:Email Verification with IYTE Domain Restriction added.

Yusuf-sudo-it commented on src/main/java/com/iyte_yazilim/proje_pazari/ProjePazariApplication.java:
```diff
@@ -2,8 +2,10 @@
+import org.springframework.scheduling.annotation.EnableAsync;
 
 @SpringBootApplication
+@EnableAsync
```
. The Best Practice: Custom Thread Pool
To make it "Production Ready," you must define a custom TaskExecutor. This limits the number of threads so your server doesn't crash under load.

Yusuf-sudo-it commented on src/main/java/com/iyte_yazilim/proje_pazari/ProjePazariApplication.java:
```diff
@@ -2,8 +2,10 @@
 
 @SpringBootApplication
+@EnableAsync
 public class ProjePazariApplication {
```
@Configuration
@EnableAsync // 1. Enables the capability
public class AsyncConfig {

    @Bean(name = "taskExecutor")
    public Executor taskExecutor() {
        ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();
        
        // 2. Customize the Pool
        executor.setCorePoolSize(5);      // Minimum threads always active
        executor.setMaxPoolSize(10);      // Max threads if queue is full
        executor.setQueueCapacity(500);   // Buffer for tasks before spawning more threads
        
        executor.setThreadNamePrefix("AsyncThread-");
        executor.initialize();
        return executor;
    }
}

neonid0 commented on src/main/java/com/iyte_yazilim/proje_pazari/domain/validation/IyteEmail.java:
```diff
@@ -7,15 +7,11 @@
-    String message() default "Invalid URL";
-
+@Constraint(validatedBy = IyteEmailValidator.class)
+public @interface IyteEmail {
```
Use valuobjects instead

neonid0 commented on src/main/java/com/iyte_yazilim/proje_pazari/domain/validation/IyteEmailValidator.java:
```diff
@@ -0,0 +1,23 @@
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.List;
+
+public class IyteEmailValidator implements ConstraintValidator<IyteEmail, String> {
```
Use valueobject

neonid0 commented on src/main/java/com/iyte_yazilim/proje_pazari/domain/entities/User.java:
```diff
@@ -11,9 +18,30 @@
 @AllArgsConstructor
 @NoArgsConstructor
 @SuppressWarnings("unused")
+@Entity
```
This is just a domain entity, not DAO.
- *In Domain-Driven Design (DDD), it's generally recommended that domain objects do not connect directly to the database. Instead, the repository pattern is used to handle data access, allowing the domain model to remain independent of the underlying data storage mechanisms.*

neonid0 commented on src/main/java/com/iyte_yazilim/proje_pazari/domain/events/UserRegisteredEvent.java:
```diff
@@ -0,0 +1,14 @@
+
+@Getter
+@AllArgsConstructor
+public class UserRegisteredEvent {
```
extend DomainEvent

neonid0 commented on src/main/java/com/iyte_yazilim/proje_pazari/domain/entities/BaseEntity.java:
```diff
@@ -1,59 +1,55 @@
+
 @Getter
 @Setter
+@MappedSuperclass
```
Do not use persistence in domain layer

neonid0 commented on src/main/java/com/iyte_yazilim/proje_pazari/application/config/ApplicationConfig.java:
```diff
@@ -1,8 +1,124 @@
+import com.iyte_yazilim.proje_pazari.application.commands.registerUser.RegisterUserValidator;
+import com.iyte_yazilim.proje_pazari.application.commands.resendVerificationEmail.ResendVerificationEmailCommand;
+import com.iyte_yazilim.proje_pazari.application.commands.resendVerificationEmail.ResendVerificationEmailHandler;
+import com.iyte_yazilim.proje_pazari.application.commands.verifyEmail.VerifyEmailCommand;
```
Fetch latest dev branch then use existing code for you solutions. ApplicationConfig is not required now.

neonid0 commented on src/main/java/com/iyte_yazilim/proje_pazari/application/services/VerificationTokenService.java:
```diff
@@ -0,0 +1,32 @@
+import java.util.UUID;
+
+@Service
+public class VerificationTokenService {
```
Services should be in application layer

neonid0 commented on src/main/java/com/iyte_yazilim/proje_pazari/infrastructure/email/EmailVerificationEventListener.java:
```diff
@@ -0,0 +1,84 @@
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class EmailVerificationEventListener {
```
Use event handlers in application layer
- *In Domain-Driven Design (DDD), event handlers are used to respond to domain events that occur within the same domain, allowing for a clear separation of concerns and better organization of code. Event listeners, on the other hand, are often associated with broader event systems and may not be as tightly coupled to the domain logic, which can lead to less clarity in handling domain-specific events.*

neonid0 commented on src/main/java/com/iyte_yazilim/proje_pazari/infrastructure/persistence/models/UserEntity.java:
```diff
@@ -49,23 +51,29 @@ public class UserEntity {
     @Column(name = "github_url")
     private String githubUrl;
 
+    @Column(name = "email_verified", nullable = false)
```
Use separate tables for those. So, we dont need to update user table with each verification req.

neonid0 commented on src/main/java/com/iyte_yazilim/proje_pazari/infrastructure/persistence/UserRepository.java:
```diff
@@ -1,21 +1,16 @@
-
     Optional<UserEntity> findByEmail(String email);
-
+    Optional<UserEntity> findByVerificationToken(String token);
```
replace this logic

neonid0 commented on src/main/java/com/iyte_yazilim/proje_pazari/ProjePazariApplication.java:
```diff
@@ -2,11 +2,14 @@
+import org.springframework.scheduling.annotation.EnableAsync;
 
 @SpringBootApplication
+@EnableAsync
```
this not necessary

neonid0 reviewed (changes requested):
#### Obey the DDD
- Firstly, fetch the latest code from dev.
- Use domain events best
- Use validators on application
- Use services on application

UlasGokkaya committed:
Conflict Error Solved.

UlasGokkaya committed:
Used valueobjects. DAO issue cleared and separate tables used too.

UlasGokkaya committed:
Domain Event extended. And MappedSuperClass, Transient annotations removed from BaseEntity to make domain layer-persistance-ignorant.

UlasGokkaya committed:
Services implemented to application layer.Move event listener to application layer as EmailVerificationEventHandler.

UlasGokkaya committed:
EnableAsync annotation removed from ProjePazariApplication file.

UlasGokkaya committed:
Last issues fixed and ApplicationConfig cleared. Because all beans are in handlers.

UlasGokkaya committed:
feat: add email verification fields and IYTE domain validation

neonid0 committed:
Merge branch 'dev' into feature/issue-15-email-verification-iyte

neonid0 committed:
refactor: simplify domain layer entities and events

neonid0 committed:
refactor: clean up register user command and handler

neonid0 committed:
refactor: improve email verification command structure

neonid0 committed:
refactor: update event handler and login handler

neonid0 committed:
refactor: update auth controller

neonid0 committed:
test: update user registration event handler tests

ErkanArikan committed:
refactor: apply code review feedback

- Replace annotation-based validation with IyteEmail Value Object
- Remove JPA annotations from domain entities
- Separate email verification into dedicated table
- Move event handlers to application layer
- Remove duplicate @EnableAsync from main class
- Clean up unnecessary ApplicationConfig

neonid0 committed:
fix: remove non-existent email verification field mappings from RegisterUserMapper

- Remove mappings for emailVerified, verificationToken, verificationTokenExpiresAt, emailVerifiedAt
- These fields were removed from User entity but mapping annotations were not cleaned up
- Fixes compilation errors in MapStruct mapper generation

neonid0 committed:
style: remove trailing whitespace in docker-compose.yml

- Clean up trailing spaces in MinIO healthcheck configuration

neonid0 committed:
refactor: clean up User entity imports and documentation

- Remove unused LocalDateTime import
- Format Javadoc comment to single line for consistency

neonid0 committed:
refactor: improve email verification implementation

- Refactor VerifyEmailHandler for better error handling and code clarity
- Update ResendVerificationEmailHandler imports
- Clean up UserRegisteredEventHandler
- Update EmailVerificationRepository query method
- Remove unused imports from ProjePazariApplication

neonid0 committed:
Merge branch 'dev' into feature/issue-15-email-verification-iyte

neonid0 committed:
fix: extend basecontroller
